pkg/controller/controller_instance.go

```diff
@@ -341,6 +370,126 @@ func (c *controller) initOrphanMitigationCondition(instance *v1beta1.ServiceInst
 	return false, nil
 }
 
+// setRetryBackoffRequired marks the specified instance/generation as needing a
+// delay before the next provision/update is attempted.  We always set this flag
+// before attempting a provision or update operation in case we must retry.  This
+// will eventually be cleared by the background worker running
+// purgeExpiredRetryEntries() or when the operation is successful.
+func (c *controller) setRetryBackoffRequired(instance *v1beta1.ServiceInstance) {
+	pcb := pretty.NewInstanceContextBuilder(instance)
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(instance)
+	if err != nil {
+		glog.Errorf(pcb.Messagef("Couldn't create a key for object %+v: %v", instance, err))
+		return
+	}
+
+	c.instanceOperationRetryQueue.mutex.Lock()
+	defer c.instanceOperationRetryQueue.mutex.Unlock()
+	retryEntry, found := c.instanceOperationRetryQueue.instances[key]
+	if !found || retryEntry.generation != instance.Generation {
+		retryEntry.generation = instance.Generation
+
+		// reset the backoff as the generation changed
+		if found {
+			c.instanceOperationRetryQueue.rateLimiter.Forget(key)
+		}
+	}
+	retryEntry.dirty = true
+	c.instanceOperationRetryQueue.instances[key] = retryEntry
+	glog.V(4).Info(pcb.Messagef("added %v generation %v to backoffBeforeRetrying map", key, instance.Generation))
+}
+
+// backoffAndRequeueIfRetrying returns true if this is a retry and a backoff
+// (delay) needs to be observed before retrying.  This only applies to
+// Provisioning and Updating and is generation specific.  If the generation has
+// been bumped since the instance was added to the retry map there will be no
+// backoff delay.
+func (c *controller) backoffAndRequeueIfRetrying(instance *v1beta1.ServiceInstance, operation string) bool {
+	pcb := pretty.NewInstanceContextBuilder(instance)
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(instance)
+	if err != nil {
+		glog.Errorf(pcb.Messagef("Couldn't create a key for object %+v: %v", instance, err))
+		return false
+	}
+	delay := time.Millisecond * 0
```

Comment:
You don't need to declare a zero `delay` var there, you could just do 
```go
delay := retryEntry.calculatedRetryTime.Sub(now) 
```
below (replace `=` with `:=`)

Reply:
agreed @nilebox, I think this was left over after a refactor and I missed it.  This was just merged, next time I'm in here I'll clean it up.